driver/src/orderbook/streamed/state.rs

```diff
@@ -159,9 +143,16 @@ impl State {
         Ok(())
     }
 
-    fn order_deletion(&mut self, event: &OrderDeletion) -> Result<()> {
-        // Orders are allowed to be deleted multiple times.
-        self.orders.remove(&(event.owner, event.id));
+    fn order_deletion(&mut self, event: &OrderDeletion, block_batch_id: BatchId) -> Result<()> {
+        if let Some(order) = self.orders.get(&(event.owner, event.id)) {
+            ensure!(
+                !order.is_valid_in_batch(block_batch_id - 1),
+                "deleting valid order"
+            );
+            self.orders.remove(&(event.owner, event.id));
```

Comment:
I think order deletion also needs a pending state. This is because an order can be valid for batch X and deleted in batch X+1, but then we will just not have the order if we try to query the orderbook at batch X.

Reply:
If we are solving batch X, then X+1 is the active batch. If the order was valid in X then it cannot get deleted in X+1 as per the `ensure` above.

Reply:
Oh yeah! It can only be deleted if its not valid in the solving batch!